lib/bigstringaf.ml

```diff
@@ -93,33 +93,28 @@ let[@inlined] get_bounds name ?(off=0) ?len t =
   then invalid_bounds name buffer_len off len
   else (off, len)
 
-let copy t ~off ~len =
-  let buffer_len = length t in
-  if len < 0 || off < 0 || buffer_len - off < len
-  then invalid_bounds "copy" buffer_len off len;
+let copy ?off ?len t =
```

Comment:
Optional arguments require allocation when the arguments are provided. Most performance-sensitive OCaml code most avoid allocation, and so it was an conscious choice to make these operations not use optional arguments.